dsc/utilities/aws/s3.py

```diff
@@ -40,22 +38,33 @@ def archive_file_with_new_key(
         )
 
     def put_file(
-        self, file_content: str | bytes, bucket: str, key: str
-    ) -> PutObjectOutputTypeDef:
-        """Put a file in a specified S3 bucket with a specified key.
+        self,
+        bucket: str,
+        key: str,
+        file_content: str | bytes = b"",
```

Comment:
I'm curious, why might we have a default for an empty byte string?

Reply:
Resolving this comment and addressing in latter comment.